docs/user-guide.md

```diff
@@ -341,6 +343,7 @@ won't need to know.
 
 There are a couple of places where the syntax of type annotations like `@Nullable`
 is counterintuitive.
+TODO: make the principle clear: type use annotation immediately before the annotated type.
 
 1. For a nested type like `Map.Entry`, if you want to say that the value can be null
```

Comment:
Actually for the array case, but I can't comment there: Should we call out the difference between declaration and type use annotations here? That is a frequent point of confusion.